src/Courses/TopicSettings/TopicSettings.tsx

```diff
@@ -39,6 +39,7 @@ export const TopicSettings: React.FC<TopicSettingsProps> = ({selected, setTopic}
     // This is a hack to allow us to update the selected TopicObject with DEF file information but not
     // lose all the user input that might be in the form.
     const [oldSelectedState, setOldSelectedState] = useState<TopicObject>(selected);
+    const [saving, setSaving] = useState<boolean>(false);
```

Comment:
I'd like to see some sort of UI component that informs the user that their update is being saved -- and that student grades are being recalculated using the new parameters... it should include a warning/note that this process can take several seconds.

Reply:
That could be a snackbar update.